### PR TITLE
8316056: Open source several Swing JTree tests

### PR DESCRIPTION
**Pull request by DamonGuy**
These are the tests being converted:

javax/swing/JTree/4210432/bug4210432.java
javax/swing/JTree/4213868/bug4213868.java
javax/swing/JTree/4224491/bug4224491.java
javax/swing/JTree/4237370/bug4237370.java
javax/swing/JTree/4662505/bug4662505.java

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316056](https://bugs.openjdk.org/browse/JDK-8316056): Open source several Swing JTree tests (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - Committer) ⚠️ Review applies to [a9bd7dcc](https://git.openjdk.org/jdk/pull/15756/files/a9bd7dcce39a6cad526806c80f203b0e89e4de0c)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15756/head:pull/15756` \
`$ git checkout pull/15756`

Update a local copy of the PR: \
`$ git checkout pull/15756` \
`$ git pull https://git.openjdk.org/jdk.git pull/15756/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15756`

View PR using the GUI difftool: \
`$ git pr show -t 15756`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15756.diff">https://git.openjdk.org/jdk/pull/15756.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15756#issuecomment-1720198709)

### CODE REVIEW

File: test/jdk/javax/swing/JTree/bug4213868.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import javax.swing.SwingUtilities;
+import javax.swing.tree.DefaultMutableTreeNode;
+
+public class bug4213868 {
+    public static JTree createTree() {
+        DefaultMutableTreeNode root =
+                new DefaultMutableTreeNode(Integer.valueOf(0), true);
+        JTree tree = new JTree(root);
+        for (int i = 1; i < 10; i++) {
+            root.add(new DefaultMutableTreeNode(Integer.valueOf(i)));
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JTree/bug4213868.java:44]:
```suggestion
                new DefaultMutableTreeNode(0, true);
        JTree tree = new JTree(root);
        for (int i = 1; i < 10; i++) {
            root.add(new DefaultMutableTreeNode(i));
```

**Review by honkar-jdk — APPROVED** [review]
Tests look good.

File: test/jdk/javax/swing/JTree/bug4237370.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            int rows = getRowCount();
+            if (rows != testMap[testIndex]) {
+                throw new RuntimeException("Bad row count: reported " + rows +
+                                " instead of " + testMap[testIndex]);
+            } else {
+                testIndex++;
+            }
+        }
+
+        public void treeExpanded (TreeExpansionEvent e) {
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JTree/bug4237370.java:56]:
```suggestion
        public void treeExpanded(TreeExpansionEvent e) {
```

Comment (DamonGuy) [reply test/jdk/javax/swing/JTree/bug4237370.java]: Fixed

File: test/jdk/javax/swing/JTree/bug4224491.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import javax.swing.SwingUtilities;
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.DefaultTreeModel;
+
+public class bug4224491 {
+    private static boolean modelChanged = false;
+    private static boolean invokesStopCellEditingChanged = false;
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
```
[the remark this answers is not in the document]
Comment (DamonGuy) [reply test/jdk/javax/swing/JTree/bug4224491.java]: Updated where applicable

File: test/jdk/javax/swing/JTree/bug4213868.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> {
+            JTree parent = createTree();
+            AccessibleContext c = parent.getAccessibleContext()
+                    .getAccessibleChild(0).getAccessibleContext();
+            if (c.getAccessibleChild(1).getAccessibleContext()
+                    .getAccessibleIndexInParent() != 1) {
```
[the remark this answers is not in the document]
Comment (DamonGuy) [reply test/jdk/javax/swing/JTree/bug4213868.java]: Added the newlines and alignments, thanks

File: test/jdk/javax/swing/JTree/bug4213868.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        return tree;
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> {
+            JTree parent = createTree();
+            AccessibleContext c = parent.getAccessibleContext()
+                    .getAccessibleChild(0).getAccessibleContext();
```
[the remark this answers is not in the document]
Comment (DamonGuy) [reply test/jdk/javax/swing/JTree/bug4213868.java]: Added the newlines and alignments, thanks

File: test/jdk/javax/swing/JTree/bug4213868.java (hunk 1 above)
Comment (DamonGuy) [reply test/jdk/javax/swing/JTree/bug4213868.java]: Changed to normal int. Thanks!

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**